// @azure/arm-subscriptions/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Workload} that the service accepts. */
export var KnownWorkload;
(function (KnownWorkload) {
    /** Production */
    KnownWorkload["Production"] = "Production";
    /** DevTest */
    KnownWorkload["DevTest"] = "DevTest";
})(KnownWorkload || (KnownWorkload = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Accepted */
    KnownProvisioningState["Accepted"] = "Accepted";
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link AcceptOwnership} that the service accepts. */
export var KnownAcceptOwnership;
(function (KnownAcceptOwnership) {
    /** Pending */
    KnownAcceptOwnership["Pending"] = "Pending";
    /** Completed */
    KnownAcceptOwnership["Completed"] = "Completed";
    /** Expired */
    KnownAcceptOwnership["Expired"] = "Expired";
})(KnownAcceptOwnership || (KnownAcceptOwnership = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link Provisioning} that the service accepts. */
export var KnownProvisioning;
(function (KnownProvisioning) {
    /** Pending */
    KnownProvisioning["Pending"] = "Pending";
    /** Accepted */
    KnownProvisioning["Accepted"] = "Accepted";
    /** Succeeded */
    KnownProvisioning["Succeeded"] = "Succeeded";
})(KnownProvisioning || (KnownProvisioning = {}));
